include/physics/diff_physics.h

```diff
@@ -240,22 +240,45 @@ class DifferentiablePhysics
    * Most derived systems will not have to reimplment this function; however
    * any system which reimplements mass_residual() may have to reimplement
    * time_evolving() to prepare data structures.
+   *
+   * This method is deprecated. Instead, use the time_evolving override
+   * and specify the order-in-time of the variable, either 1 or 2. This method
+   * assumes the variable is first order for backward compatibility.
    */
   virtual void time_evolving (unsigned int var) {
+    libmesh_deprecated();
     if (_time_evolving.size() <= var)
-      _time_evolving.resize(var+1, false);
-    _time_evolving[var] = true;
+      _time_evolving.resize(var+1, 0);
+    _time_evolving[var] = 1;
   }
 
   /**
+   * Tells the DiffSystem that variable var is evolving with
+   * respect to time.  In general, the user's init() function
+   * should call time_evolving() with order 1 for any variables which
+   * behave like du/dt = F(u), with order 2 for any variables that
+   * behave like d^2u/dt^2 = F(u), and should not call time_evolving()
+   * for any variables which behave like 0 = G(u).
+   *
+   * Most derived systems will not have to reimplment this function; however
+   * any system which reimplements mass_residual() may have to reimplement
+   * time_evolving() to prepare data structures.
+   */
+  virtual void time_evolving (unsigned int var, unsigned int order);
+
+  /**
    * Returns true iff variable \p var is evolving with
    * respect to time.  In general, the user's init() function
    * should have set time_evolving() for any variables which
    * behave like du/dt = F(u), and should not call time_evolving()
    * for any variables which behave like 0 = G(u).
    */
   bool is_time_evolving (unsigned int var) const {
-    return _time_evolving[var];
+    libmesh_assert_less(var,_time_evolving.size());
+    libmesh_assert( _time_evolving[var] == 0 ||
+                    _time_evolving[var] == 1 ||
+                    _time_evolving[var] == 2 );
+    return !(_time_evolving[var] == 0);
```

Comment:
For that matter, why not just return the unsigned int?  That will be API compatible with 99% of uses in theory and will probably be more useful in practice.